src/main/java/hudson/tasks/junit/PackageResult.java

```diff
@@ -148,15 +145,7 @@ public int getSkipCount() {
 
     @Override
     public Object getDynamic(String name, StaplerRequest req, StaplerResponse rsp) {
-        JunitTestResultStorage storage = JunitTestResultStorage.find();
-        ClassResult result;
-        if (!(storage instanceof FileJunitTestResultStorage)) {
-            Run<?, ?> run = Stapler.getCurrentRequest().findAncestorObject(Run.class);
-            TestResultImpl pluggableStorage = storage.load(run.getParent().getFullName(), run.getNumber());
-            result = pluggableStorage.getClassResult(name);
-        } else {
-            result = getClassResult(name);
-        }
+        ClassResult result = getClassResult(name);
```

Comment:
get by package result also loads the full class result, this code before was ignoring the package and only worked in tests because we had a Klazz and a other.Klazz. and so the wrong Klazz was being picked.

Now we load from the correct package